### ISSUE TITLE
nginx won't start, release amd64-0.34.2-ls31

### ISSUE THREAD

**Issue by azinchen** [linuxserver/docker-letsencrypt#313]
Two issues.

First, If I upgrade letsencrypt image to the latest version in container with already preconfigured /config volume, I get
```
[services.d] done.
Server ready
nginx: [alert] detected a LuaJIT version which is not OpenResty's; many optimizations will be disabled and performance will be compromised (see https://github.com/openresty/luajit2 for OpenResty's LuaJIT or, even better, consider using the OpenResty releases from https://openresty.org/en/download.html)
nginx: [error] lua_load_resty_core failed to load the resty.core module from https://github.com/openresty/lua-resty-core; ensure you are using an OpenResty release from https://openresty.org/en/download.html (rc: 2, reason: module 'resty.core' not found:
	no field package.preload['resty.core']
	no file './resty/core.lua'
	no file '/usr/share/luajit-2.1.0-beta3/resty/core.lua'
	no file '/usr/local/share/lua/5.1/resty/core.lua'
	no file '/usr/local/share/lua/5.1/resty/core/init.lua'
	no file '/usr/share/lua/5.1/resty/core.lua'
	no file '/usr/share/lua/5.1/resty/core/init.lua'
	no file '/usr/share/lua/common/resty/core.lua'
	no file '/usr/share/lua/common/resty/core/init.lua'
	no file './resty/core.so'
	no file '/usr/local/lib/lua/5.1/resty/core.so'
	no file '/usr/lib/lua/5.1/resty/core.so'
	no file '/usr/local/lib/lua/5.1/loadall.so'
	no file './resty.so'
	no file '/usr/local/lib/lua/5.1/resty.so'
	no file '/usr/lib/lua/5.1/resty.so'
	no file '/usr/local/lib/lua/5.1/loadall.so')
nginx: [warn] could not build optimal variables_hash, you should increase either variables_hash_max_size: 1024 or variables_hash_bucket_size: 64; ignoring variables_hash_bucket_size
```

Second, I tried to create clean letsencrypt container. My docker-compose.yml file
```
version: '3'

services:
  letsencrypt:
    image: linuxserver/letsencrypt:latest
    restart: always
    environment:
      - EMAIL=*
      - URL=*
      - SUBDOMAINS=test
      - VALIDATION=dns
      - DNSPLUGIN=cloudflare
      - STAGING=true
    volumes:
      - ./volumes/letsencrypt:/config
```
The ./volumes/letsencrypt did not exist. Then I ran container, stopped it, updated volumes/letsencrypt/dns-conf/cloudflare.ini file, then started it again. Now I get
```
[services.d] done.
nginx: [emerg] PEM_read_bio_DHparams("/config/nginx/dhparams.pem") failed (SSL: error:0909006C:PEM routines:get_name:no start line:Expecting: DH PARAMETERS)
Server ready
nginx: [emerg] PEM_read_bio_DHparams("/config/nginx/dhparams.pem") failed (SSL: error:0909006C:PEM routines:get_name:no start line:Expecting: DH PARAMETERS)
nginx: [emerg] PEM_read_bio_DHparams("/config/nginx/dhparams.pem") failed (SSL: error:0909006C:PEM routines:get_name:no start line:Expecting: DH PARAMETERS)
nginx: [emerg] PEM_read_bio_DHparams("/config/nginx/dhparams.pem") failed (SSL: error:0909006C:PEM routines:get_name:no start line:Expecting: DH PARAMETERS)
```
Last line repeats forever.

**Comment by azinchen** [comment]
Please fix it ASAP, all my services are down right now.

**Comment by thelamer** [comment]
If all my Infrastructure was down due to an upgrade I would revert to the previous working version. Not demand open source developers fix it ASAP. 
Lua errors have been reported across different infra in multiple issues in this repo. 
We are waiting on an upstream fix for this https://github.com/linuxserver/docker-letsencrypt/issues/303

**Comment by aptalca** [comment]
Lua error is harmless unless you use Lua (I doubt it). In any case, the error does not prevent nginx from starting up.

Variables_hash is also a warning, does not prevent nginx from starting up.

You are not passing PUID and PGID or the TZ.

And finally, it looks like your dhparams is corrupt, perhaps you interrupted the generation process?

Delete that and restart, then watch the logs, it might take a while to regenerate.

But as thelamer said earlier, you're better off creating a container with your previous data.

Since this is a configuration issue and not a bug with the image (apart from the upstream issue where is another issue for it), I will close this. If you need further support, you visit our forum or discord.
